unit_tests/engine/test_rule_loader.cpp

```diff
@@ -0,0 +1,235 @@
+#include <gtest/gtest.h>
+
+#include "falco_engine.h"
+#include "rule_loader_reader.h"
+#include "rule_loader_compiler.h"
+
+class engine_loader_test : public ::testing::Test {
```

Comment:
Very neat setup @LucaGuerra much more elegant :heart: than what I had tried in the past :woman_facepalming:

Reply:
I noticed we actually do something similar in at least three different locations in the test suite (including this one). This is out of scope for this PR but we need to consolidate them and have one fixture to allow loading engine + rulesets.

Reply:
Big +1, it would be super useful to do this cleanup!